# Introduction to Long-Read Data Analysis


This page provides entry-level bioinformatic tutorials for Long-Read data analysis with particular focus on Oxford Nanopore data. The material was initially created for the *[Long-read, long-reach sequencing workshop](https://cba.anu.edu.au/news-events/long-read-long-reach-sequencing-workshop)* held at the Australian National University, 11-15th of November 2019 but has been extended and will be updated for new courses or whenever I find the time.

## Content
 * [Tutorial Overview](OV.md)
 * [Set-Up](SU.md)
   * [Create a VirtualBox Environment](SU_VB.md)
   * [Install Ubuntu in your Virtual machine](SU_U.md)
   * [Add the Guest Additions to VirtualBox](SU_GA.md)
   * [Run the install script for tools and packages](SU_I.md)
   * [Download the tutorial data](SU_D.md)
 * [Basecalling using Guppy](BS_G.md)
 * [Quality control](QC.md)
   * [FastQC](QC_F.md)
   * [PycoQC](QC_P.md)
   * [MinIONQC](QC_M.md)
 * [Read filtering, trimming and adapter removal](FTR.md)
   * [Porechop](FTR_P.md)
   * [NanoFilt](FTR_N.md)
 * [Genome Assembly](ASS.md)
   * [Minimap - Miniasm](ASS_M.md)
   * [Flye](ASS_F.md)
   * [Shasta](ASS_S.md)
 * [Consensus Sequences & Error correction](www.test.com)
   * [Racon](ECR_R.md)
   * [Minipolish](ECR_MI.md)
   * [Medaka](ECR_ME.md)
   * [Pilon](ECR_P.md)
 * [Variant Calling](VC.md)
   * [Variant Calling using Sniffles](VS_S.md)
 * [Assembly Improvements using Bandage](BAN.md)
 * [Appendix](APP.md)
   * [Frequently Asked Questions](FAQs.md)
   * [List of Tutorial Software](APP_TOOLS.md) 
   * [Tutorial Data & Installation Script](APP_DATA.md)
   * [File Formats](APP_FORM.md)
   * [Metrics](APP_STAT.md)
